### PR TITLE
Smarter custom properties with Houdini’s new API

### LINKED ISSUE #1122: content: Blog post for CSS Properties and Values API L1
**Author**: @Snugug
**Reviewer**: @jpmedley

**A one to two sentence description of your post**
A basic how-to article describing the CSS Properties and Values API Level 1 (Houdini), which is an API that expands the existing CSS custom properties to improve usability of other Houdini APIs.

**Target publish date:** `2019-09-19`
- [x] Check this box if this is a hard deadline.

**Chrome Status entry:** https://www.chromestatus.com/feature/5640265926705152

**Process**
Make a copy of [this template](https://docs.google.com/document/d/1ydauCufwwavStaKxhIDHuNivVxgQBOdDno4uLjGIjmE/edit)
and fill in the proposal and outline sections. When you're finished, drop a link
to the doc in the 1st step below and check the box. A member of the web.dev team
will assign this ticket to themselves and work as your reviewer to help you
complete the remaining steps.

- [x] 1. I've done my proposal https://docs.google.com/document/d/14QeJKjqVV0nD3w0S6E0Afcimp041cEeYaMMdSIsYfRU/edit?usp=sharing
- [x] 2. Proposal approved — _reviewer checks this_
- [x] 3. I've added a draft to my proposal doc
- [x] 4. Draft approved — _reviewer checks this_
- [x] 5. I've submitted a pull request `#1497`
- [ ] 6. Pull request approved — _reviewer checks this_

### PR DESCRIPTION
**Pull request by jpmedley**
Fixes #1122 

Note: If this is written content for the site please include the issue number from your [content proposal](https://github.com/GoogleChrome/web.dev/issues?q=is%3Aissue+is%3Aopen+label%3A%22content+proposal%22).

Changes proposed in this pull request:

- New blog article describing CSS Properties and Values Level 1.
- 
-

### CODE REVIEW

File: src/site/content/en/blog/css-props-and-vals/index.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  --stop-color: green;
+}
+```
+
+This will animate our custom property that's part of the `linear-gradient`, thus
+animating our linear gradient. Check out the Glitch below to see the full code
+in action and play around with it yourself.
+
+<!-- Copy and Paste Me -->
+<div class="glitch-embed-wrap" style="height: 420px; width: 100%;">
```
Comment (robdodson) [inline src/site/content/en/blog/css-props-and-vals/index.md]: btw the third panel in the glitch look like it's had some of its vertical content cut off for me:

image

File: src/site/content/en/blog/css-props-and-vals/index.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+API](https://developers.google.com/web/updates/2018/01/paintapi) already shipped
+and now Custom Props and Vals, our creative toolbox is expanding, allowing us to
+define typed CSS properties and use them to create and animate new and exciting
+designs. There's more on the way, too, in the [Houdini issue
+queue](https://github.com/w3c/css-houdini-drafts/issues) where you can give
+feedback and see what's next for Houdini. Houdini exists to develop features
+that explain the "magic" of styling and layout on the web, so get out there and
+put those magical features to good use.
+
+Photo by Maik Jonietz on Unsplash
```
Comment (robdodson) [inline src/site/content/en/blog/css-props-and-vals/index.md]:
```suggestion
_Photo by Maik Jonietz on Unsplash._
```

Can you link to the image and link to unsplash? See the example at the bottom of this post: https://web.dev/samesite-cookies-explained

File: src/site/content/en/blog/css-props-and-vals/index.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+Now, the custom property `--my-color` is registered as a color! This tells the
+browser what kinds of values are allowed and how it can type and treat that
+property!
+
+### Anatomy of a registered property
+
+`registerProperty()` takes four options:
+
+<dl>
```
Comment (robdodson) [inline src/site/content/en/blog/css-props-and-vals/index.md]: The definition list doesn't look that nice when it renders. I think an alternative might be to use h4 headings and to move the types next to the property name, similar to how you might do it in something like TypeScript:

```
#### `name: string`
The name of the custom property.
  
#### `syntax: string`
How to parse the custom property. You can find a complete list of possible values in the <a href="https://drafts.csswg.org/css-values-3/">CSS Values and Units</a> specification. Defaults to <code>*</code>.

#### `inherits: boolean`
Whether it inherits its parent’s value. Defaults to <code>true</code>.

#### `initialValue: string`
Initial value of the custom property.
```

Here's what that looks like:

image

There's a tiny bit of left margin from having a `<code>` in the header. I can do a follow up PR to remove that from our styles.

File: src/site/content/en/blog/css-props-and-vals/index.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+  <dd>The name of the custom property. String.</dd>
+  <dt><code>syntax</code></dt>
+  <dd>How to parse the custom property. You can find a complete list of possible values in the <a href="https://drafts.csswg.org/css-values-3/">CSS Values and Units</a> specification. Defaults to <code>*</code>. String.</dd>
+  <dt><code>inherits</code></dt>
+  <dd>Whether it inherits its parent’s value. Defaults to <code>true</code>. Boolean.</dd>
+  <dt><code>initialValue</code></dt>
+  <dd>Initial value of the custom property. String.</dd>
+</dl>
+
+Registering a property looks like this:
```
Comment (robdodson) [inline src/site/content/en/blog/css-props-and-vals/index.md]: Just curious if you might want to put this before the text that appears at line 79. I noticed that line 79 starts in with `registerProperty()` but that's the first time it's mentioned in the article so it is sort of lacking a bit of context.

Another option would be to add a sentence before line 79 that says something like, "To register a property, use the `registerProperty()` method in your JavaScript. `registerProperty()` takes four options:..."

File: src/site/content/en/blog/css-props-and-vals/index.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+CSS custom properties, also known as [CSS
+variables](https://developers.google.com/web/updates/2016/02/css-variables-why-should-you-care),
+let you define your own properties in CSS and use their values throughout your
+CSS. While incredibly useful today, they have shortcomings that can make them
+hard to work with: they can take any value so they may be accidentally
+overridden with something unexpected, they always inherit their values from
+their parent, and you can't transition them. With Houdini's [CSS Properties and
+Values API Level 1](https://drafts.css-houdini.org/css-properties-values-api/),
+now available in Chrome 78, ßthese shortcomings are transcended, making CSS
```
[the remark this answers is not in the document]
Comment (jpmedley) [reply src/site/content/en/blog/css-props-and-vals/index.md]: This is my fat fingering. Removed.

[the remark this answers is not in the document]
Comment (jpmedley) [reply src/site/content/en/blog/css-props-and-vals/index.md]: In Visual Studio Code, I keep hitting option+s to save my work when I mean to hit command+s. It adds that symbol.

File: src/site/content/en/blog/css-props-and-vals/index.md (hunk 1 above)
Comment (jpmedley) [reply src/site/content/en/blog/css-props-and-vals/index.md]: @Snugug FYI

File: src/site/content/en/blog/css-props-and-vals/index.md (hunk 2 above)
Comment (jpmedley) [reply src/site/content/en/blog/css-props-and-vals/index.md]: This was throwing: `warning  Don’t use literal URLs without angle brackets  no-literal-urls  remark-lint`. There was nothing in it that wasn't in the samesite cookies source. I used the `--no-verify` flag to submit.

**Comment by jpmedley** [comment]
PTAL

**Comment by jpmedley** [comment]
@robdodson For some reason, web.dev doesn't like the links in the last paragraph of the markdown. I can't figure out what's throwing it off.  It doesn't have anything in it (as far as I can tell) that's not in https://web.dev/samesite-cookies-explained, which is already published.

File: src/site/content/en/blog/css-props-and-vals/index.md (hunk 2 above)
Comment (robdodson) [reply src/site/content/en/blog/css-props-and-vals/index.md]: I think it throws that warning if you paste the actual URL on the page without putting it inside of a markdown link.

Comment (robdodson) [reply src/site/content/en/blog/css-props-and-vals/index.md]: Took a look at the PR and I agree that seems weird. You can add a line above that snippet to tell the linter to chill out:

```
<!--lint disable no-literal-urls-->
```

I can look into why it's doing that later.

**Comment by jpmedley** [comment]
What line to disable the linter?

**Comment by jpmedley** [comment]
Found it. Never mind.

**Pull request merged**